### ISSUE TITLE
Method make

### ISSUE THREAD

**Issue by danielbohnrs** [spatie/enum#54]
I was looking for a method like `findByIndex` or something like that. After looking at the code, I find out that the method `make` does not create a new enum, only gets a existing one by its index or name/value (at the docs its says that `make`: "Creating an enum from a value"). 
Am I the only one that find this a little confusing?

**Comment by Gummibeer** [comment]
Hey, thanks for reporting.
We won't change the `make()` method name. But I'm fully open for an improved wording in the docs!
Something like new instance of it or similar.
Primary these small things are always hard to detect if you've write the whole for days/weeks and everything is clear.^^

**Comment by AbdullahFaqeir** [comment]
Hello,

If you can provide some extra explanation, I’d love to work on it

**Comment by danielbohnrs** [comment]
I have an Enum like this one below.
I will save only the index of the selected option by the user. 
I was looking for a way to get a specific Enum by its index.
My first approach was to clone the private function `resolveByIndex` into my Enum, to use:
TipoOcorrencia::resolveByIndex($index);
Then I saw that this same function is used by method `make`.
So I realized that if I use the method `make` it would have the same effect...
But it was not intuitive in the docs because it says 'Creating a enum', but by this way I'm not creating a new one, only getting a existing one.
Do you got my point of view?

```
class TipoOcorrencia extends Enum
{
    public static function abastecimento(): TipoOcorrencia
    {
        return new class() extends TipoOcorrencia {
            public function getValue(): string
            {
                return 'Abastecimento';
            }
            public function getIndex(): int
            {
                return 0;
            }
        };
    }
    
    public static function inicioViagem(): TipoOcorrencia
    {
        return new class() extends TipoOcorrencia {
            public function getValue(): string
            {
                return 'Inicio Viagem';
            }
            public function getIndex(): int
            {
                return 1;
            }
        };
    }
}
```

**Comment by AbdullahFaqeir** [comment]
According to what I’ve understood from your elaboration, we can either do as @gummibeer suggested give a new description to the ‘’make()’’ method, or we can just simply add a new methods to resolve the enum by index or value, which actually will alias the ‘’resloveByName()’’ and ‘’resolveByIndex()’’. 

Am I right??

**Comment by Gummibeer** [comment]
Because the core problem here was the not intuitive wording in the docs I would change it.

> Create a new instance of the enum by passed in value or index.

Would this be clear enough? 🤔

**Comment by AbdullahFaqeir** [comment]
@Gummibeer I guess it’s clear now regarding the new wording, right @danielbohrs?

And regarding the naming, I didn’t suggest a name, I said the new methods would only alias the current resolvers but I didn’t mention their new names, yet I agree with the naming you suggested. If all good, I’d submit a PR today.

**Comment by Gummibeer** [comment]
@AbdullahFaqeir could you send two PRs? The wording changes in the docs should be fast and easy to merge. The new methods I would like to wait for @brendt feedback. So could be that we decide for another naming.
Thanks for your time in advance! 😊

**Comment by AbdullahFaqeir** [comment]
Sure!

**Comment by danielbohnrs** [comment]
> @Gummibeer I guess it’s clear now regarding the new wording, right @danielbohrs ?
> 
> And regarding the naming, I didn’t suggest a name, I said the new methods would only alias the current resolvers but I didn’t mention their new names, yet I agree with the naming you suggested. If all good, I’d submit a PR today.

@AbdullahFaqeir yes, now it seems clear to me.
Thanks for your attention!

**Comment by Gummibeer** [comment]
Will not be implemented because of #56 and the simplification of this package.
